framework/src/problems/FEProblemBase.C

```diff
@@ -2929,7 +2929,10 @@ FEProblemBase::addFVInterfaceKernel(const std::string & fv_ik_name,
                                     const std::string & name,
                                     InputParameters & parameters)
 {
-  addObject<FVInterfaceKernel>(fv_ik_name, name, parameters);
+  /// We assume that variable1 and variable2 can live on different systems, in this case
+  /// the user needs to create two interface kernels with flipped variables and parameters
```

Comment:
if this is always true, we should check for it
maybe create an issue to keep track of some multi-system diagnostics we can think of

Reply:
I agree that an issue would be good. Right now if they don't add an interface kernel for "var2", then there's probably nothing telling the user they they are not getting interface kernel contributions for "var2". Perhaps a resolution for the issue would be to behind the scenes add both interface kernels automatically, swapping var1 and var2, when we detect that the two variables have different systems

Reply:
I don't like the behind the scenes option. I tried it and I think it can lead to confusion. Especially if we have custom parameters that belong to var1 and var2. We cannot flip those together with the variables in a general way. I'll add an issue but for now the two signs of not adding the right amount of interface kernels are:
(1) my added documentation to interface kernels which I'll do today
(2) the obviously wrong results or convergence issues (which are not the nicest indicators for input error)

Reply:
FY information, the issue is created:
https://github.com/idaholab/moose/issues/25622